nucleus/common/common-util/src/main/java/com/sun/enterprise/util/io/FileUtils.java

```diff
@@ -174,7 +174,7 @@ public static boolean deleteFileMaybe(File f) {
     ///////////////////////////////////////////////////////////////////////////
 
     public static boolean safeIsDirectory(File f) {
-        return (f != null && f.exists() && f.isDirectory());
+        return !(f == null || !f.exists() || !f.isDirectory());
```

Comment:
This is not easier to understand. Does sonar suggest to change it like that?

Reply:
Never mind, didn't see the full diff. I still suggest to express this with `&&` instead.

Reply:
I agree. If Sonar did not like it, it will not like this even more. 
The original if was better, I would only remove those brackets and also f.exists, because f.isDirectory returns true only if file exists. That is probably why Sonar complained:
```
     * @return <code>true</code> if and only if the file denoted by this
     *          abstract pathname exists <em>and</em> is a directory;
     *          <code>false</code> otherwise
```

Reply:
Weird ... I'm receiving yesterday e-mails now :dagger: